exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -691,57 +746,198 @@ func (ku *Kucoin) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Subm
+				case order.Buy:
+					stopOrderBoundary = "down"
+				}
+			case order.TrailingStop, order.Stop:
+				switch s.Side {
+				case order.Sell:
+					// Stop-loss when order type is order.Stop
+					stopOrderBoundary = "down"
+				case order.Buy:
+					// Take Profit when order type is order.Stop
+					stopOrderBoundary = "up"
+				}
+			}
+		}
+		o, err = ku.PostFuturesOrder(ctx, &FuturesOrderParam{
 			ClientOrderID: s.ClientOrderID,
 			Side:          sideString,
 			Symbol:        s.Pair,
 			OrderType:     s.Type.Lower(),
 			Size:          s.Amount,
 			Price:         s.Price,
-			StopPrice:     s.TriggerPrice,
 			Leverage:      s.Leverage,
+			VisibleSize:   0,
 			ReduceOnly:    s.ReduceOnly,
 			PostOnly:      s.PostOnly,
-			Hidden:        s.Hidden})
+			Hidden:        s.Hidden,
+			Stop:          stopOrderBoundary,
+			StopPrice:     s.TriggerPrice,
+			StopPriceType: stopOrderType,
+		})
 		if err != nil {
 			return nil, err
 		}
 		return s.DeriveSubmitResponse(o)
 	case asset.Spot:
-		timeInForce := ""
-		if s.Type == order.Limit {
-			switch {
-			case s.FillOrKill:
-				timeInForce = "FOK"
-			case s.ImmediateOrCancel:
-				timeInForce = "IOC"
-			case s.PostOnly:
+		switch {
+		case ku.HFSpot:
+			o, err = ku.HFSpotPlaceOrder(ctx, &PlaceHFParam{
+				ClientOrderID: s.ClientOrderID,
+				Symbol:        s.Pair,
+				OrderType:     s.Type.Lower(),
+				Side:          s.Side.String(),
+				Price:         s.Price,
+				Size:          s.Amount,
+				PostOnly:      s.PostOnly,
+				Hidden:        s.Hidden,
+				Iceberg:       s.Iceberg,
+			})
+			if err != nil {
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(o)
+		default:
+			switch s.Type {
+			case order.Limit, order.Market, order.StopLimit, order.StopMarket:
+				var oType order.Type
+				switch s.Type {
+				case order.Limit:
+					oType = order.Limit
+				case order.Market:
+					oType = order.Market
+				}
+				var timeInForce string
+				if oType == order.Limit {
+					switch {
+					case s.FillOrKill:
+						timeInForce = "FOK"
+					case s.ImmediateOrCancel:
+						timeInForce = "IOC"
+					case s.PostOnly:
+					default:
+						timeInForce = "GTC"
+					}
+				}
+				var stopType string
+				var stopPrice float64
+				switch {
+				case s.RiskManagementModes.StopLoss.Enabled && s.RiskManagementModes.StopEntry.Enabled:
+					return nil, errors.New("can not enable more than one risk management")
+				case s.RiskManagementModes.StopEntry.Enabled:
+					stopType = "entry"
+					stopPrice = s.RiskManagementModes.StopEntry.Price
+				case s.RiskManagementModes.StopLoss.Enabled:
+					stopType = "loss"
+					stopPrice = s.RiskManagementModes.StopLoss.Price
+				}
+				var o string
+				if stopType != "" {
+					o, err = ku.PostStopOrder(ctx,
+						s.ClientOrderID,
+						sideString,
+						s.Pair.String(),
+						oType.Lower(), "", stopType, "", s.TradeMode,
+						timeInForce, s.Amount, s.Price, stopPrice, 0,
+						0, 0, s.PostOnly, s.Hidden, s.Iceberg)
```

Comment:
I don't understand how Iceberg is only in this function. They can be for limit or stop orders.
https://www.kucoin.com/docs/rest/spot-trading/orders/place-order

Reply:
That was my mistake